File: Leetcode/array_hashing/python3/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        return sorted(s) == sorted(t)


#
# 64 ms
# 17.6 MB

# class Solution:
#    def isAnagram(self, s: str, t: str) -> bool:
#        sCount, tCount = Counter(s), Counter(t)
#        return sCount == tCount

File: Leetcode/array_hashing/scala/valid-anagram.scala
object Solution {
    def isAnagram(s: String, t: String): Boolean = {
        if (s.length != t.length) {
            return false
        } 

        val charsCount = Array.fill(26)(0)

        val toIdx = (_: Char) - 'a'

        for (c <- s) {
            charsCount(toIdx(c)) += 1
        }

        for (c <- t) {
            charsCount(toIdx(c)) -= 1
        }

        charsCount.forall(_ == 0)
    }
}